utils/data-file.c

```diff
@@ -469,8 +495,14 @@ int open_info_file(struct opts *opts, struct uftrace_data *handle)
 	handle->last_perf_idx = -1;
 	INIT_LIST_HEAD(&handle->events);
 
-	if (fread(&handle->hdr, sizeof(handle->hdr), 1, fp) != 1)
-		pr_err("cannot read header data");
+	if (is_info_txt) {
+		if (read_header(&handle->hdr, fp) < 0)
+			pr_err("cannot read header data");
+	}
+	else {
+		if (fread(&handle->hdr, sizeof(handle->hdr), 1, fp) != 1)
+			pr_err("cannot read header data");
+	}
```

Comment:
It's just a refactoring request. I think we can make `read_header` to handle both `info.txt` and `info` inside. So we can change the signature as follows.
```c
int read_header(struct uftrace_file_header *hdr, FILE *fp, bool is_info_txt);
```
Then handle `fread` inside of `read_header`.